### PR TITLE
Filtering for Contacts, Group Chat Filtering, and clean up Column types

### LINKED ISSUE #39: Add Filtering Functions to TopFriends Query
Add the logic to filter on the TopFriends query just like the WordOrEmoji query.

### PR DESCRIPTION
**Pull request by alexdanilowicz**
- Add filtering logic for contacts, group chat filtering
- Add OutputColumns to make it clear what columns can be referenced. 

Resolves #39

### CODE REVIEW

File: electron/src/chatBro/queries/TopFriends/columns.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export const Columns = {
```
Comment (Teddarific) [inline electron/src/chatBro/queries/TopFriends/columns.ts:1]: Use enums here

File: electron/src/chatBro/queries/TopFriends/TopFriends.ts [hunk 2]
```diff
@@ -1,62 +1,70 @@
...
+    -- NOTE: filters should always be applied as earliest as possible
+    ${allFilters}
   GROUP BY h.id, is_from_me
 `;
+};
 
-function getSentOrRecieved(alias: string, isFromMeToggle: string) {
+function getSentOrRecieved(
+  alias: string,
+  isFromMeToggle: string,
```
Comment (ayc19) [inline electron/src/chatBro/queries/TopFriends/TopFriends.ts:26]: Think it'd be better to have isFromMeToggle just be isFromMe and as a boolean, and convert to the string within the function. Seems kinda awk to explicitly pass in '0' as the user

File: electron/src/chatBro/queries/TopFriends/filters.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+function contactFilter(filters: TopFriendsTypes.Filters): string | undefined {
+  if (_.isEmpty(filters.contact)) {
+    return undefined;
+  }
+  return `${Columns.FRIEND} = "${filters.contact}"`;
+}
+
+function groupChatFilter(filters: TopFriendsTypes.Filters): string | undefined {
+  if (filters.groupChat !== undefined) {
+    return `message.cache_roomnames ${
```
Comment (ayc19) [inline electron/src/chatBro/queries/TopFriends/filters.ts:21]: what is cache_roomnames

File: electron/src/chatBro/queries/TopFriends/filters.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+function wordFilter(filters: TopFriendsTypes.Filters): string | undefined {
+  if (_.isEmpty(filters.word)) {
+    return undefined;
+  }
+  // NOTE: using LIKE because TopFriends query is not split word by word
+  return `LOWER(${Columns.TEXT}) LIKE "%${filters.word?.toLowerCase()}%"`;
+}
+
+function contactFilter(filters: TopFriendsTypes.Filters): string | undefined {
```
Comment (ayc19) [inline electron/src/chatBro/queries/TopFriends/filters.ts:12]: minor nit: Think the filters should just have what's needed for each specific one vs passing in all filters.

**Review by ayc19 — CHANGES_REQUESTED** [review]

File: electron/src/chatBro/queries/TopFriends/filters.ts (hunk 3 above)
Comment (alexdanilowicz) [reply electron/src/chatBro/queries/TopFriends/filters.ts]: group chats

File: electron/src/chatBro/queries/TopFriends/TopFriends.ts (hunk 2 above)
Comment (alexdanilowicz) [reply electron/src/chatBro/queries/TopFriends/TopFriends.ts:26]: Fair enough. Thanks, changed.

File: electron/src/chatBro/queries/TopFriends/filters.ts (hunk 4 above)
Comment (alexdanilowicz) [reply electron/src/chatBro/queries/TopFriends/filters.ts:13]: Typescript is complaining and I can't seem to figure out how to fix it, let's table this and fix it later.

**Pull request merged**